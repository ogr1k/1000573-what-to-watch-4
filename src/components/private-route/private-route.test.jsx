import React from "react";
import renderer from "react-test-renderer";
import PrivateRoute from "./private-route.jsx";
import {BrowserRouter} from "react-router-dom";

const mockProps = {
  "exact": true,
  "path": `/films/:id/review`,
  "authorisationStatus": `AUTH`,
  "location": {
    "pathname": `/films/3/review`,
    "search": ``,
    "hash": ``,
    "key": `h7dlyc`
  },
  "computedMatch": {
    "path": `/films/:id/review`,
    "url": `/films/3/review`,
    "isExact": true,
    "params": {
      "id": `3`
    }
  }
};

const MockComponent = () => {
  return (
    <div>
    </div>
  );
};


it(`PrivateRoute renders component`, () => {
  const tree = renderer
    .create(
        <BrowserRouter>
          <PrivateRoute {...mockProps} component={MockComponent}
          />
        </BrowserRouter>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
