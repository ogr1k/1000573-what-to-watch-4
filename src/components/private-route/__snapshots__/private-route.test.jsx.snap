// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute renders component 1`] = `<div />`;
